vcs/git/repo.go

```diff
@@ -235,11 +233,19 @@ func (r *Repository) Commits(opt vcs.CommitsOptions) ([]*vcs.Commit, uint, error
 			commits = append(commits, r.makeCommit(c))
 		}
 		total++
-		return total < uint(MaxCommits)
```

Comment:
Ok. Do you know what clients use the `git` backend? I'm only aware of ones that use `gitcmd` backend at this time.

Once they're updated to use `NoTotal`, this performance hack can be reverted. Note that it technically results in incorrect behavior, since if you ask for more than all commits, you'll get only 250 commits max, even if there are more (and of course total will also be capped to 250).